polyfill.js

```diff
@@ -20,7 +20,25 @@
   } catch (e) {
     var CustomEvent = function(event, params) {
       var evt, origPrevent;
-      params = params || {};
+
+      // We use here some version of `Object.assign` implementation, to create a shallow copy of `params`.
+      // Based on https://github.com/christiansany/object-assign-polyfill/blob/213cc63df14515fb543117059d1576204bfaa8a7/index.js
+      params = (function(from) {
```

Comment:
Let's name this new variable `newParams` to be clear - and also because it's cleaner to not overwrite the input parameter.